walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,103 @@
 
 package com.google.sps;
 
+import java.util.*;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Comparator;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> event_times = new ArrayList<TimeRange>();
+    List<TimeRange> times_mandatory = new ArrayList<TimeRange>();    
+    long duration = request.getDuration();
+    List<TimeRange> results = new ArrayList<TimeRange>();
+    if((int) duration > TimeRange.END_OF_DAY - TimeRange.START_OF_DAY){
+      return results;
+    }
+    Collection<String> attendees = request.getAttendees();
+    Collection<String> optional_attendees = request.getOptionalAttendees();
+    int e_count = 0;
+    for(Event event : events){
+      TimeRange event_time = event.getWhen();
+      Collection<String> event_people = event.getAttendees();
+      boolean people_in_both = false;
```

Comment:
this naming is a bit unclear, can you add a comment describing what "both" means or maybe just change the name of the bool (even if it makes it longer)